src/common/UtilAll.cpp

```diff
@@ -267,6 +267,50 @@ string UtilAll::getHomeDirectory() {
   return homeDir;
 }
 
+static bool createDirectoryInner(const char *dir)
+{
+    if (dir == nullptr)  {
+        return false;
+    }
+    if (access(dir, 0) == -1) {
+#ifdef _WIN32
+        int flag = mkdir(dir);
+#else
+        int flag = mkdir(dir, 0755);
+#endif
+        if (flag == 0) {
```

Comment:
I think use "return flag == 0;" directly is better, and 'ret' is better than 'flag'.